easybuild/easyblocks/generic/intelbase.py

```diff
@@ -131,10 +131,9 @@ def parse_components_list(self):
                                      % self.cfg['components'])
         else:
             self.install_components = []
-            for comp in available_components:
-                for comp_regex in self.cfg['components']:
-                    if re.search(comp_regex, comp):
-                        self.install_components.append(comp)
+            for comp_regex in self.cfg['components']:
+                comps = [comp for comp in available_components if re.match(comp_regex, comp)]
```

Comment:
with `search` you may have matches you don't want, for example with a pattern like `comp`

with `search`, all component names that have `comp` in them would be select

with `match`, you should use `comp.*`, and then you'll only get stuff that starts with `comp`, so it gives you more control